tests/swagger/test_resources.py

```diff
@@ -1,104 +1,44 @@
-import json
+import unittest
+import six
 
-from bravado_core.request import unmarshal_request
 from bravado_core.response import validate_response
+from bravado_core.param import marshal_param
 
-from .support import (SwaggerTest, MINIMALIST_BUCKET, MINIMALIST_GROUP,
-                      MINIMALIST_COLLECTION, MINIMALIST_RECORD)
+from .support import SwaggerTest
 
 
-class SwaggerResourcesTest(SwaggerTest):
+class MetaResourcesTest(type):
+    """Metaprogram test cases for easy debugging."""
 
-    allowed_failures = ['version']
+    def __init__(cls, *args, **kargs):
+        unittest.case.TestCase = cls
+        cls.generate_test_resources()
 
-    def test_existing_path(self):
-        for resource in self.resources.values():
+    def generate_test_resources(cls):
+        for resource in cls.resources.values():
             for op_id, op in resource.operations.items():
-                self.setUp()
-                self.request.path = {
-                    'bucket_id': 'b1',
-                    'group_id': 'g1',
-                    'collection_id': 'c1',
-                    'record_id': 'r1'
-                }
-                resp = self.validate_request_call(op, expect_errors=True)
-
-                if op_id not in self.allowed_failures:
-                    # Method must exist (404 and 405 means it might not exist)
-                    self.assertIn(resp.status_int, [200, 201, 400])
-
-    def test_resource_utilities(self):
-        resource = self.resources['Utilities']
-        for op_id, op in resource.operations.items():
-                if op_id not in self.allowed_failures:
-                    self.validate_request_call(op)
-
-    def test_resource_batch(self):
-        resource = self.resources['Batch']
-        for op in resource.operations.values():
-            self.setUp()
-            requests = [{'path': '/v1/buckets'}]
-            defaults = {'method': 'POST'}
-            self.request.json = lambda: dict(
-                requests=requests,
-                defaults=defaults
-            )
-            self.validate_request_call(op)
-
-    def test_resource_buckets(self):
-        resource = self.resources['Buckets']
+                if op_id not in cls.allowed_failures:
+                    setattr(cls, 'test_resource_%s' % op_id,
+                            lambda self: cls.validate_request_call(self, op))
```

Comment:
So what I was thinking when I suggested this was something a little different:

```python
def operation_test_method(op):
    def test_method(self):
        pass   # basically "validate_request_call"
    return test_method

def resource_test_class(resource):
    attrs = {}
    for op_id, op in resource.operations.items():
        attrs["test_{}".format(op_id)] = operation_test_method(op)
    return type("{}Test".format(resource.name), (SwaggerResourceTest,), attrs)

BucketsTest = resource_test_class(resources['Buckets'])   # or whatever the syntax is
CollectionsTest = resource_test_class(resources['Collections'])    # etc.
```

This approach means you don't have to explicitly handle metaclasses, which could be easier to read. Another approach might be to use a class decorator:

```python
def operation_test_method(op):
    def method(self):
        pass  # again, basically validate_request_call

def resource_test_case(resource):
    def decorator(cls):
        for op_id, op in resource.operations.items():
            setattr(cls, "test_{}".format(op_id), operation_test_method(op))
        return cls

    return decorator

@resource_test_case(resources['Buckets'])
class BucketsTest(SwaggerTest):
    pass
```

But having reread this a bunch of times, I think the hardest part to read was just the diff. Reading the new file, it isn't that confusing, except for a couple of quirks in how I think you are interfacing with Bravado.